### ISSUE TITLE
Installer does not support nativeSystemd

### ISSUE THREAD

**Issue by TerrorJack** [nix-community/NixOS-WSL#179]
## Bug description
`nativeSystemd` doesn't seem to work yet.

## To Reproduce

Specify `wsl.nativeSystemd = true;` in `configuration.nix`, build `nixos-wsl-installer.tar.gz`, import and run it.

**Expected behavior**
The installer script should start.

## Logs
```
PS C:\Users\terrorjack\Downloads> wsl -d NixOS
Catastrophic failure
Error code: Wsl/Service/CreateInstance/E_UNEXPECTED
```
```
PS C:\Users\terrorjack\Downloads> wsl --version
WSL version: 1.0.3.0
Kernel version: 5.15.79.1
WSLg version: 1.0.47
MSRDC version: 1.2.3575
Direct3D version: 1.606.4
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.900
```

**Comment by SuperSandro2000** [comment]
Can you reproduce this when changing in on an already installed system?

**Comment by TerrorJack** [comment]
When `wsl.nativeSystemd = false` or the line removed, the first launch is stuck infinitely after `14.34 MiB freed by hard-linking 6765 files` and didn't even make it to `Starting systemd...`. This is the tarball built with `main` branch and `nixos-unstable`, I'll try other configurations later.

**Comment by SuperSandro2000** [comment]
> the first launch is stuck infinitely after `14.34 MiB freed by hard-linking 6765 files` and didn't even make it to `Starting systemd...`.

Does it start after your restart wsl?

**Comment by TerrorJack** [comment]
After restarting it's stuck after:

```
setting up /etc...
setting up /bin...
```

First installation seems to work after switching to `nixos-22.11` :)

**Comment by TerrorJack** [comment]
Switching to `wsl.nativeSystemd = true` after installation also seems to work. Feel free to close this ticket and thanks a lot for your awesome work! It would be nice to check invalid fields at installer build-time though.

**Comment by shyim** [comment]
I also have the same issue on first boot. Didn't changed Anything. Just run the import and it's broken

```
<3>WSL (8) ERROR: CreateProcessEntryCommon:559: chdir(/root) failed 2
Unpacking root file system...
 307MiB 0:00:08 [34.9MiB/s] [========================================================================>] 100%
Activating nix configuration...
Copying /usr/share/applications
Copying /usr/share/icons
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
setting up /etc...
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
setting up /bin...
Cleaning up installer files...
finding garbage collector roots...
removing stale link from '/nix/var/nix/gcroots/auto/6yfbrv8wfkyl40wq6827lamr9m29z7wg' to '/build/nix/var/nix/profiles/per-user/root/channels-1-link'
removing stale link from '/nix/var/nix/gcroots/auto/zxi8v969la5n36bg7gs016wgcif4qjhp' to '/build/nix/var/nix/profiles/system-1-link'
deleting garbage...
deleting '/nix/store/5mism3qrfc3ah9ahvf7sczxy4bax9cgq-pv-1.6.20'
deleting '/nix/store/0p9v7yvblgxsds9pbsrir3af6gw0x77s-busybox-1.35.0'
deleting '/nix/store/cq6l0b0iv2y3gln3xi6nzxs57mz5rxws-tarball'
deleting '/nix/store/ab5mixyq1zg0x5lnr5i7m4l5815fp1h8-wsl-prepare'
deleting '/nix/store/scr6balqjknqpgm7igd0b4pagmp2ldbl-installer.sh'
deleting '/nix/store/pmb55f6rk98cdycqf589vb3xpbg15k28-source'
deleting unused links...
note: currently hard linking saves -0.00 MiB
6 store paths deleted, 309.05 MiB freed
Optimizing store...
5.74 MiB freed by hard-linking 3714 files
Starting systemd...
```

**Comment by shyim** [comment]
Imported directly the other file like described in https://github.com/nix-community/NixOS-WSL/pull/143 and it works lol

**Comment by nzbr** [comment]
@shyim Your problem wasn't related to native systemd, but another known bug with the container-based implementation, that should be fixed with the next release. See also #156

**Comment by SuperSandro2000** [comment]
> Schwerwiegender Fehler

translation: critical error

**Comment by nzbr** [comment]
Possibly. What does `wsl --version` output?

**Comment by 573** [comment]
@nzbr 
>➜  wsl --version
WSL-Version: 1.1.0.0
Kernelversion: 5.15.83.1
WSLg-Version: 1.0.48
MSRDC-Version: 1.2.3770
Direct3D-Version: 1.608.2-61064218
DXCore-Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows Version: 10.0.19045.2486

**Comment by nzbr** [comment]
@K900 is this the version that's affected by #199?

**Comment by K900** [comment]
Yes.

**Comment by shadowrylander** [comment]
So I seem to be getting the same problem after building the tarball from the master branch on the following `wsl` version:
```
WSL version: 1.1.3.0
Kernel version: 5.15.90.1
WSLg version: 1.0.49
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22000.1696
```

**Comment by nzbr** [comment]
@shadowrylander did you modify configuration.nix? If yes, what changes did you make?

**Comment by shadowrylander** [comment]
No changes other than adding `nativeSystemd = true;`.

**Comment by nzbr** [comment]
Yeah, that's the problem: The installer does not support nativeSystemd yet. You'll need to disable it, install and then re-enable it

**Comment by shadowrylander** [comment]
I enabled it and rebuilt it the first time, but `/sbin/init` apparently linked to a wsl shim or something like that? Is it not supposed to be pointing to `systemctl` or similar?

**Comment by nzbr** [comment]
Normally, it's linked to systemd, but NixOS needs to execute the activation scripts first. So we have the shim that does all the setup tasks and then execs systemd

**Comment by K900** [comment]
Fixed in 23.05.
